### PR TITLE
[ShapeOpt] use kratos logger

### PR DESCRIPTION
**Pull request by armingeiser**
replace all `std::cout` and `print()` with kratos logger

Currently i used `ShapeOptimization` as label. This is up to discussion (see comments below).
The output of one step shell optimization e.g. looks like this:

```log

ShapeOptimization: =================================================================== 
ShapeOptimization:  Tue Sep 24 13:43:10 2019 : Starting optimization iteration  7 
ShapeOptimization: ===================================================================
 

ShapeOptimization: Starting to update the mesh 
ShapeOptimization: Time needed for updating the mesh =  0.0 s 
::[KSM Simulation]:: : STEP:  7 
::[KSM Simulation]:: : TIME:  7.699999999999999 

> Starting primal analysis for response: strain_energy 
> Time needed for solving the primal analysis: 0.13 s 
> Time needed for calculating the response value: 0.0 s 

> Starting gradient calculation for response: strain_energy 
> Time needed for calculating gradients: 0.02 s 
ShapeOptimization: Starting to update mapper...
ShapeOptimization: Finished updating of mapper in 0.000171402 s.

ShapeOptimization: Starting inverse mapping of DF1DX...
ShapeOptimization: Finished mapping in 3.6213e-05 s.

ShapeOptimization: No constraints given or active. The negative objective gradient is chosen as search direction...

ShapeOptimization: Starting mapping of CONTROL_POINT_UPDATE...
ShapeOptimization: Finished mapping in 3.2298e-05 s.

ShapeOptimization: Current value of objective =   1.11393E-07 
ShapeOptimization: Absolut change of objective =  -9.88432E+01  [%] 
ShapeOptimization: Relative change of objective =  -5.56567E-01  [%]
 

ShapeOptimization: Time needed for current optimization step =  0.331 s 
ShapeOptimization: Time needed for total optimization so far =  2.434 s 

ShapeOptimization: Optimization problem converged within a relative objective tolerance of  1.0 %. 
::[KSM Simulation]:: : Analysis -END-  

ShapeOptimization: ============================================================================================================== 
ShapeOptimization: Finished optimization                                                                                            
ShapeOptimization: ==============================================================================================================
```

Note: the lines without label come from the response function.

### CODE REVIEW

File: applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h [hunk 1]
```diff
@@ -94,10 +94,10 @@ class DampingUtilities
           mMaxNeighborNodes( DampingSettings["max_neighbor_nodes"].GetInt() )
     {
         BuiltinTimer timer;
-        std::cout << "\n> Creating search tree to perform damping..." << std::endl;
+        KRATOS_INFO("") << "\n> Creating search tree to perform damping..." << std::endl;
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h]: why don't you add a label? Makes it easier to track where the messages come from

Comment (loumalouomega) [reply applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h]: Idem

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h]: @dbaumgaertner what label would you prefer?

```
> Creating search tree to perform damping...
```
becomes
```
ShapeOptimization: Creating search tree to perform damping...
```
or
```
ShapeOpt: Creating search tree to perform damping...
```
or 
```
KSO: Creating search tree to perform damping...
```

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h]: i would for `ShapeOptimization` or `ShapeOpt`

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h]:
```
ShapeOptimizationApplication: Creating search tree to perform damping...
```
is to long

**Comment by philbucher** [comment]
Just to add that often the class or fct name is used to have a more specific output
Up to you, just saying

**Comment by armingeiser** [comment]
I see. For warning messages i think that's fine, but for info IMO it is not relevant. Let's wait what @dbaumgaertner has to say.

If we go for class names we should at least make sure that all labels of the application have the same lenght. Otherwise the output becomes very hard to read.

File: applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h [hunk 2]
```diff
@@ -508,7 +508,7 @@ class MapperVertexMorphing : public Mapper
             double sum_of_weights = 0.0;
 
             if(number_of_neighbors >= max_number_of_neighbors)
-                std::cout << "\n> WARNING!!!!! For node " << node_i.Id() << " and specified filter radius, maximum number of neighbor nodes (=" << max_number_of_neighbors << " nodes) reached!" << std::endl;
+                KRATOS_WARNING("ShapeOptimization") << "For node " << node_i.Id() << " and specified filter radius, maximum number of neighbor nodes (=" << max_number_of_neighbors << " nodes) reached!" << std::endl;
```
Comment (dbaumgaertner) [inline applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h]: In such cases I would specify the class or function name as label, to allow for an improved debugging.

**Review by dbaumgaertner — CHANGES_REQUESTED** [review]
Looks good from my side. To the label:

Unless it is a warning, a general optimization label is enough. The given one is definitely way too long. ShapeOpt is better. So I suggest to update the Warnings such that they include the class or function name as label and for all the others use "KratosShape". What do you think @armingeiser ?

Just wondering. Is it also possible to add a time-stamp additionally to or instead of the label? Might be a good instrument for an improved analysis of the computation time.

File: applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h (hunk 2 above)
Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h]: Will add a more detailed label for warnings.

**Comment by armingeiser** [comment]
>  The given one is definitely way too long. ShapeOpt is better. ... use "KratosShape". 

KratosShape has a good length, i just think that having the word Kratos in every line is a bit overkill because it is quite clear that Kratos is running if one launches it. However thats at least a label that we also used in some presentations etc.

ShapeOpt is more general but not nice to read.

> Just wondering. Is it also possible to add a time-stamp additionally to or instead of the label? Might be a good instrument for an improved analysis of the computation time.

I think this is possible. e.g. 
`12:05:01: Start initializing mappers`
`KSO - 12:05:01: Start initializing mappers`
`KratosShape - 12:05:01: Start initializing mappers`

`12:05:01: Start initializing mappers`
`12:05:01 - KSO: Start initializing mappers`
`12:05:01 - KratosShape: Start initializing mappers`

**Comment by philbucher** [comment]
`KSO` sounds strange to me, not really descriptive

maybe the timestamp could even be integrated in the Logger (long-term solution)

**Comment by dbaumgaertner** [comment]
Oh, I mixed the words sorry. I wanted to actually confirm your suggestion with "ShapeOpt". "KratosShape" is indeed a bit overkill. Actually also ShapeOpt is also somewhat clunky. Having said this, we need a better name for the App!

About the time. Let's go for your last suggestion: 

`12:05:01 - ShapeOpt: Start initializing mappers`

I think it's nice to have a bit of feedback about the time, what do you think?

**Comment by armingeiser** [comment]
Good for me. 
I like the time stamp - we have it in our other internal software and its very useful.
As @philbucher suggests i will have a look at the time stamp in the logger - in a different PR.

**Comment by armingeiser** [comment]
@philbucher is there a global settings object in json for the logger?

**Comment by philbucher** [comment]
quite sure there is not

**Comment by armingeiser** [comment]
```
=============================================================================== 
ShapeOpt:  Thu Sep 26 11:27:55 2019 : Starting optimization iteration  7 
===============================================================================
 
 
ShapeOpt: Starting to update the mesh 
ShapeOpt: Time needed for updating the mesh =  0.0 s 
::[KSM Simulation]:: : STEP:  7 
::[KSM Simulation]:: : TIME:  7.699999999999999 

> Starting primal analysis for response: strain_energy 
> Time needed for solving the primal analysis: 0.01 s 
> Time needed for calculating the response value: 0.0 s 

> Starting gradient calculation for response: strain_energy 
> Time needed for calculating gradients: 0.01 s 
ShapeOpt: Starting to update mapper...
ShapeOpt: Finished updating of mapper in 0.000108737 s.

ShapeOpt: Starting inverse mapping of DF1DX...
ShapeOpt: Finished mapping in 2.3445e-05 s.

ShapeOpt: No constraints given or active. The negative objective gradient is chosen as search direction...

ShapeOpt: Starting mapping of CONTROL_POINT_UPDATE...
ShapeOpt: Finished mapping in 3.3776e-05 s.
 
ShapeOpt: Current value of objective =   1.11393E-07 
ShapeOpt: Absolut change of objective =  -9.88432E+01  [%] 
ShapeOpt: Relative change of objective =  -5.56567E-01  [%]
 
 
ShapeOpt: Time needed for current optimization step =  0.035 s 
ShapeOpt: Time needed for total optimization so far =  0.265 s 
 
ShapeOpt: Optimization problem converged within a relative objective tolerance of  1.0 %. 
::[KSM Simulation]:: : Analysis -END-  
 
=============================================================================== 
ShapeOpt: Finished optimization 
===============================================================================
```

And with the other PR for the logger


```
=============================================================================== 
11:26:52 - ShapeOpt:  Thu Sep 26 11:26:52 2019 : Starting optimization iteration  7 
===============================================================================
 
 
11:26:52 - ShapeOpt: Starting to update the mesh 
11:26:52 - ShapeOpt: Time needed for updating the mesh =  0.0 s 
11:26:52 - ::[KSM Simulation]:: : STEP:  7 
11:26:52 - ::[KSM Simulation]:: : TIME:  7.699999999999999 
11:26:52 - 
> Starting primal analysis for response: strain_energy 
11:26:52 - > Time needed for solving the primal analysis: 0.07 s 
11:26:52 - > Time needed for calculating the response value: 0.0 s 
11:26:52 - 
> Starting gradient calculation for response: strain_energy 
11:26:52 - > Time needed for calculating gradients: 0.01 s 
11:26:52 - ShapeOpt: Starting to update mapper...
11:26:52 - ShapeOpt: Finished updating of mapper in 0.000109317 s.

11:26:52 - ShapeOpt: Starting inverse mapping of DF1DX...
11:26:52 - ShapeOpt: Finished mapping in 2.2546e-05 s.

11:26:52 - ShapeOpt: No constraints given or active. The negative objective gradient is chosen as search direction...

11:26:52 - ShapeOpt: Starting mapping of CONTROL_POINT_UPDATE...
11:26:52 - ShapeOpt: Finished mapping in 1.9605e-05 s.
 
11:26:52 - ShapeOpt: Current value of objective =   1.11393E-07 
11:26:52 - ShapeOpt: Absolut change of objective =  -9.88432E+01  [%] 
11:26:52 - ShapeOpt: Relative change of objective =  -5.56567E-01  [%]
 
 
11:26:52 - ShapeOpt: Time needed for current optimization step =  0.189 s 
11:26:52 - ShapeOpt: Time needed for total optimization so far =  1.383 s 
 
11:26:52 - ShapeOpt: Optimization problem converged within a relative objective tolerance of  1.0 %. 
11:26:52 - ::[KSM Simulation]:: : Analysis -END-  
 
=============================================================================== 
11:26:52 - ShapeOpt: Finished optimization 
===============================================================================
```

File: applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h [hunk 3]
```diff
@@ -137,7 +137,8 @@ class MapperVertexMorphing : public Mapper
             Initialize();
 
         BuiltinTimer timer;
-        KRATOS_INFO("\nShapeOptimization") << "Starting mapping of " << rOriginVariable.Name() << "..." << std::endl;
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h]: this doesn't work?

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h]: It does, but if there is a time stamp in front the output looks strange:
```
11:26:52 - 
ShapeOpt: Optimization problem converged within a relative objective tolerance of  1.0 %. 
```

**Comment by armingeiser** [comment]
@dbaumgaertner can you approve?

**Comment by dbaumgaertner** [comment]
The output above is nice, but I think it has some minor inconsistencies here an there. Also, I could not reproduce the combined output with the time stamp. Lets quickly discuss this offline before proceeding.

**Comment by armingeiser** [comment]
The time stamp output is done in a separate merge request #5676.

> Lets quickly discuss this offline before proceeding.
:+1:

**Comment by armingeiser** [comment]
Info level output is disabled in tests. 
The labels for the structural responses are added in #5722.

File: applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py [hunk 4]
```diff
@@ -16,7 +18,7 @@
 class ShapeOptimizationTestFactory(kratos_unittest.TestCase):
     # --------------------------------------------------------------------------
     def setUp(self):
-        pass
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py:19]: I recommend to use the following:
https://github.com/KratosMultiphysics/Kratos/blob/4c2a6a31b6f2900e6d973ce60bdff5dac9268315/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py#L35-L39

out of experience this makes it much easier for debugging a single test ;)

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py:19]: All the tests can be run as standalone (without the unit test wrapper). So debugging a single test is already by default outputting everything.

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py:19]: But thanks for the hint!

Comment (philbucher) [reply applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py:19]: ah ok then please neglect my comment ;)

**Review by dbaumgaertner — APPROVED** [review]

**Pull request merged**